lib/los_santos.rb

```diff
@@ -2,19 +2,18 @@
 require 'yaml'
 
 module LosSantos
-  def self.name_day(date_string = Time.now.strftime('%d-%m'))
-    data = YAML.load_file('lib/los_santos/data/fr.yml')
+  ALLOWED_CLASS = [String, Date, Time, DateTime].freeze
+
+  def self.name_day(date = Time.now)
     begin
-      date_split = date_string.split('-')
-      return data[date_split[1].to_i][date_split[0].to_i]
-    rescue NoMethodError
-      return print_usage
+      date = Date.strptime(date, '%d-%m') if date.class == String
+    rescue ArgumentError
+      raise ArgumentError, "String usage: LosSantos.name_day 'dd-mm'"
     end
-  end
-
-  private
-
-  def self.print_usage
-    puts 'Usage: LosSantos.name_day [\'dd-mm\']'
+    unless ALLOWED_CLASS.include? date.class
+      raise ArgumentError, 'Only String, Date, Time and DateTime are allowed'
```

Comment:
Use `ALLOWED_CLASS ` variable to make this string dynamic:
`raise ArgumentError, "Only #{ALLOWED_CLASS.join(' ,')} are allowed"`

Reply:
:ok:

Reply:
This check should be the first check.